teamA/jinwoo/app.js

```diff
@@ -33,6 +33,143 @@ app.get("/ping", (req, res) => {
   res.status(200).json({ message : "pong" });
 })
 
+// 전체 게시글 조회하기
+app.get("/posts", async (req, res) => {
+  await mysqlDataSource.query(
+    `SELECT
+        usersWesta.id as userId,
+        usersWesta.profile_image as userProfileImage,
+        postsWesta.id as postingId,
+        postsWesta.title as postingTitle,
+        postsWesta.content as postingContent
+      FROM usersWesta
+      INNER JOIN postsWesta
+      ON postsWesta.user_id = usersWesta.id
+      `
+    ,(err, rows) => {
+        res.status(200).json(rows);
+      })
+});
+
+// 유저의 게시글 조회하기
+app.get("/posts/:userId", async (req, res, next) => {
+  const userId = req.params.userId;
+
+  const userInfo = await mysqlDataSource.query(
+    `SELECT
+        usersWesta.id as userId,
+        usersWesta.profile_image
+      FROM usersWesta
+      WHERE id = ${userId};
+    `);
+    
+    const posting = await mysqlDataSource.query(
+      `SELECT
+        postsWesta.id as postingId,
+        postsWesta.title as postingTitle,
+        postsWesta.content as postingContent
+      FROM postsWesta
+      INNER JOIN usersWesta
+      ON usersWesta.id = postsWesta.user_id
+      `
+    );
+      userInfo[0].postings = posting;
+      res.status(200).json({ "data" : userInfo[0] });
+    });
+
+// 유저 회원가입 하기
+app.post("/signup", async (req, res, next) => {
+  const { name, email, profileImage, password } = req.body
+  await mysqlDataSource.query(
+    `INSERT INTO usersWesta(
+      name,
+      email,
+      profile_image,
+      password
+    )
+    VALUES (?, ?, ?, ?)
+    `,
+    [ name, email, profileImage, password ]
+  );
+  res.status(201).json({ message : "Successfully created" });
+})
+
+// 게시물 등록하기
+app.post("/postup", async (req, res) => {
+  const { title, content, userId } = req.body
+  await mysqlDataSource.query(
+    `INSERT INTO postsWesta(
+      title,
+      content,
+      user_id
+    )
+    VALUES (?, ?, ?);
+    `,
+    [ title, content, userId ]
+  );
+
+  res.status(201).json({ message : "postCreated" });
+})
+
+// 좋아요 누르기
+app.post("/like", async (req, res) => {
+  const { userId, postId } = req.body
+  await mysqlDataSource.query(
+    `INSERT INTO likesWesta(
+      user_id,
+      post_id
+    )
+    VALUES (?, ?);
+    `,
+    [ userId, postId ]
+  );
+  
+  res.status(201).json({ message : "likeCreated" })
+})
+
+// 게시글 수정하기
+app.patch("/postsmodify/:postId", async (req, res, next) => {
+  const postId = req.params.postId
+  const { title, content } = req.body
+
+  await mysqlDataSource.query(
+    `UPDATE postsWesta
+    SET
+      title = ?,
+      content = ?
+    WHERE
+      id = ${postId}
+    `,
+    [ title, content ]
+  );
```

Comment:
```suggestion
  await mysqlDataSource.query(
    `UPDATE postsWesta
    SET
      title = ?,
      content = ?
    WHERE
      id = ?
    `,
    [ title, content, postId ]
  );
```

위와 같이 typeorm의 initialize를 통해 만들어진 객체의 .query method에는 SQL 문이 들어갈 수 있습니다.

그 안에 작성되는 SQL 문에는 진우님이 작성해주신 것과 같이 자리 표시자(${ })를 통해 값을 넣을 수도, 혹은 제가 suggestion으로 작성드린 SQL 안에 ? 를 지정하여 동적으로 쿼리문이 수행될 수 있도록 하는 Placeholder를 사용하는 방법으로 값을 넣을 수도 있습니다.

두가지 방법 모두 다 SQL 안에 입력값을 넣어줄 수 있습니다.

다만 제가 작성한 것과 같이 ?를 활용하여 Placeholder를 사용하는 것을 **지향**해야 하는 이유는
보안적인 측면과 관련있습니다.

백엔드는 비단 클라이언트와 데이터베이스 사이의 접점이라고 볼 수 있는 API를 단순 동작하도록 구현하는 것 뿐만이 아닌, Data를 주고 받을 때, Database에서 원하는 data를 추출할 때를 포함한 여러 과정에서의 보안성도 고려해야 합니다.

해당 리뷰와 가장 관련이 있는 악성 공격으로는 SQL Injection 이라는 것이 있고 이는 정상적인 사용자가 아닌 누군가가 SQL문에 중요한 데이터를 추출할 수 있는 값을(ex. 항상 TRUE가 될 수 있는) 의도적으로 삽입 하여 data를 탈취하는 공격이고, `${}`를 사용하면 이에 취약한 코드가 된다고 볼 수 있습니다.

따라서 가능하면 ?를 사용하고 placeholder를 통해 값을 넣어주는 방식으로 구현해주세요!